src/formatDate.js

```diff
@@ -50,7 +50,66 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const [year, fromYear, toYear, indexOfYearFrom]
+    = findYears(dateArr, fromFormat, toFormat);
+
+  if (fromYear.length > toYear.length) {
+    dateArr[indexOfYearFrom] = year.toString().slice(2);
+  }
+
+  if (fromYear.length < toYear.length && year < 30) {
+    dateArr[indexOfYearFrom] = '20' + year;
+  }
+
+  if (fromYear.length < toYear.length && year >= 30) {
+    dateArr[indexOfYearFrom] = '19' + year;
+  }
+
+  return combineDates(dateArr, fromFormat, toFormat);
+}
+
+function combineDates(date, fromFormat, toFormat) {
+  const res = [];
+  let joinWith;
+
+  for (const el of toFormat) {
+    if (!'DYM'.includes(el[0])) {
+      joinWith = el;
+
+      continue;
+    }
+
+    res.push(date[findPartDate(fromFormat, el[0])]);
+  }
+
+  return res.join(joinWith);
+}
+
+function findPartDate(fromFormat, partDate) {
+  for (let i = 0; i < 4; i++) {
+    if (fromFormat[i][0] === partDate) {
+      return i;
+    }
+  }
+}
+
+function findYears(date, fromFormat, toFormat) {
+  const years = [];
+
+  for (let i = 0; i < 4; i++) {
+    if (fromFormat[i][0] === 'Y') {
+      years[0] = date[i];
+      years[1] = fromFormat[i];
+      years[3] = i;
+    }
+
+    if (toFormat[i][0] === 'Y') {
+      years[2] = toFormat[i];
+    }
+  }
```

Comment:
very confusing code.
I don't sure that you need to create a lot of functions for this task... it's really difficult for reading

Reply:
creating many function actually make code more readible. There is a general guideline in the Ruby community that a function should ideally be short and focused, typically no more than 10-15 lines long about js dont know